students/ArchWu/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,90 @@
+"""
+    Norton Furniture project
+    store details of customers
+    store and retrieve a customer’s credit limit
+    produce monthly counts of the total number of active customers
+"""
+import logging
+import peewee
+
+logging.basicConfig(filename='basic.log', filemode='w',
+                    format='%(asctime)s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S',
+                    level='DEBUG')
+database = peewee.SqliteDatabase("customers.db")
+database.connect()
+database.execute_sql('drop table if exists customer;')
+
+class BaseModel(peewee.Model):
+    """Init of base model"""
+    class Meta:
+        """Init of Meta class"""
+        database = database
+
+class Customer(BaseModel):
+    """Customer database init"""
+    customer_id = peewee.CharField(primary_key=True, max_length=30)
+    name = peewee.CharField(max_length=50)
+    lastname = peewee.CharField(max_length=50, null=False)
+    address = peewee.CharField(max_length=75)
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320)
+    status = peewee.CharField(max_length=10)
+    credit_limit = peewee.FloatField()
+
+database.create_tables([Customer])
+
+def add_customer(customer_id, name, lastname, address, phone_number, email, status, credit_limit):
+    """add a customer"""
+    try:
+        with database.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+            )
+            customer.save()
+            logging.info("added customer %s", customer_id)
+
+    except Exception as e:
+        logging.error("Error adding customer %s", customer_id)
+
+def delete_customer(customer_id):
+    """Delete a customer of given customer_id"""
+    Customer.delete().where(Customer.customer_id == customer_id).execute()
```

Comment:
It is a good idea to determine if this user exists before trying to run the delete. That way you can throw an error and let the User know that this user does not exist